core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/arrayfunction/ArrayFunctionExpression.kt

```diff
@@ -0,0 +1,60 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.arrayfunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.ValidType
+
+sealed class ArrayFunctionExpression<T : ValidType> : TypeExpression<ArrayType<T>>, FunctionOperator {
+    private val symbol: String
+    private val array: TypeExpression<ArrayType<T>>
+    private val args: Array<out TypeExpression<out ValidType>>
+    private val extra: TypeExpression<out ValidType>?
+
+    constructor(
+        symbol: String,
+        array: TypeExpression<ArrayType<T>>,
+        vararg args: TypeExpression<out ValidType>,
+        extra: TypeExpression<out ValidType>? = null,
+    ) {
+        this.symbol = symbol
+        this.array = array
+        this.args = args
+        this.extra = extra
+    }
+
+    constructor(symbol: String, array: TypeExpression<ArrayType<T>>, vararg args: TypeExpression<ArrayType<out ValidType>>) {
+        this.symbol = symbol
+        this.array = array
+        this.args = args
+        extra = null
+    }
```

Comment:
Are you sure about the way you are doing a secondary constructor here? 
I would split the second constructor into multiple lines anyway, looks cleaner.

Reply:
It's not even needed anymore with my current implementation, so I removed it and made a primary constructor.